aten/src/ATen/native/native_functions.yaml

```diff
@@ -1466,6 +1466,12 @@
     CompositeImplicitAutograd: chunk
     NestedTensorCPU, NestedTensorCUDA: chunk_nested_tensor
 
+- func: chunk_backward(Tensor[] grads, Tensor input, int chunks, int dim) -> Tensor
+  variants: function
+  dispatch:
+    CompositeImplicitAutograd: chunk_backward
+    NestedTensorCPU, NestedTensorCUDA: chunk_backward
+
```

Comment:
If we only handle the all grads non-None case, it sounds like we can avoid this native function. We could just have a function in torch/csrc/autograd/FunctionsManual.cpp that loops through grads to verify non-None and then calls into cat.

Reply:
Just added the support for the cases when some grad are none too.

Reply:
I think we can still do it without native functions. Since we are saving self, you could reapply chunk, then zeros_like the ones corresponding to None.

Reply:
I see. If we want to support backward for chunk(dim=0) eventually, we will need to do it through native functions right?

Reply:
Actually the better way is to create a torch.zeros_like(self), chunk, copy_ each grad into the chunk outputs. We should do that in this PR today if possible since its more efficient. This would also avoid the issue of the cat(dim=0) producing a brand new offsets.

Reply:
@soulitzer , zeros_like is a good point.

But re: making a new op: in order to know where to copy_ the grads, don't you need to be able to see the offsets, which you'd need to do in ops.py?

Reply:
chunk(dim=0) should return a bunch of views, these views correspond one-to-one with the grad outputs you receive, s.t. the shape of the view is exactly the same as the shape of the grad. So all you need to do is `chunk(dim=0)[i].copy_(grad_outputs)`.

Reply:
Thanks for the suggestion, it is definitely a cleaner approach. 
One question about the copy_: We don't have copy_ support for NJT yet, I can add it but I am wondering how we should check the offsets of the self and src. I think the offsets need to be the same, but not necessary the exact same tensor. But today we can only check if the offsets are the exact same tensor in NJT, which doesn't work for the chunk backend usage. 
This won't be an issue if we implement this in ops.py since we can just copy_ the values.

Reply:
That's a good point. I wonder if we could eventually have some kind of new copy_unsafe_ op, whose semantics are that of copy_ except we ignore the shapes (safe in this context because autograd has already validated the shapes).
We'd be able to reuse this when we later add dim=0 support to unbind_backward_nested_jagged which @jbschlosser has also already implemented with the replay view + copy_ approach.

Another option is to eventually define the union find merge operations as an aten ops so that we can access them in derivatives. But seems a little wasteful because as soon as we return from this backward op, we'll never see the offsets of the replayed view again.